packages/caliper-fabric/lib/adaptor-versions/v1/fabric-gateway-v1.js

```diff
@@ -1497,73 +1481,69 @@ class Fabric extends BlockchainInterface {
      * @return {Promise<TxStatus>} The result and stats of the transaction invocation.
      * @async
      */
-    async _submitGatewayTransaction(context, invokeSettings) {
+    async _performGatewayTransaction(context, invokeSettings, isSubmit) {
 
         // Retrieve the existing contract and a client
-        const contract = await this._getUserContract(invokeSettings.invokerIdentity, invokeSettings.chaincodeId);
-        const client = this.clientProfiles.get(invokeSettings.invokerIdentity);
-
-        // Build the Caliper TxStatus, this is a reduced item when compared to the low level API capabilities
-        const txIdObject = client.newTransactionID();
-        const txId = txIdObject.getTransactionID();
-        let invokeStatus = new TxStatus(txId);
-        invokeStatus.Set('request_type', 'transaction');
-
-        if(context.engine) {
-            context.engine.submitCallback(1);
-        }
-
-        try {
-            const result = await contract.submitTransaction(invokeSettings.chaincodeFunction, ...invokeSettings.chaincodeArguments);
-            invokeStatus.result = result;
-            invokeStatus.verified = true;
-            invokeStatus.SetStatusSuccess();
-            return invokeStatus;
-        } catch (err) {
-            logger.error(`Failed to submit transaction [${invokeSettings.chaincodeFunction}] using arguments [${invokeSettings.chaincodeArguments}],  with error: ${err.stack ? err.stack : err}`);
-            invokeStatus.SetStatusFail();
-            invokeStatus.result = [];
-            return invokeStatus;
-        }
-    }
-
-    /**
-     * Submit a transaction using a Gateway contract
-     * @param {object} context The context previously created by the Fabric adapter.
-     * @param {ChaincodeQuerySettings} querySettings The settings associated with the transaction evaluation.
-     * @return {Promise<TxStatus>} The result and stats of the transaction invocation.
-     * @async
-     */
-    async _evaluateGatewayTransaction(context, querySettings) {
+        const smartContract = await this._getUserContract(invokeSettings.invokerIdentity, invokeSettings.chaincodeId);
 
-        // Retrieve the existing contract and a client
-        const contract = await this._getUserContract(querySettings.invokerIdentity, querySettings.chaincodeId);
-        const client = this.clientProfiles.get(querySettings.invokerIdentity);
+        // Create a transaction
+        const transaction = smartContract.createTransaction(invokeSettings.chaincodeFunction);
 
         // Build the Caliper TxStatus, this is a reduced item when compared to the low level API capabilities
-        const txIdObject = client.newTransactionID();
-        const txId = txIdObject.getTransactionID();
-        let invokeStatus = new TxStatus(txId);
-        invokeStatus.Set('request_type', 'query');
+        const invokeStatus = new TxStatus(uuid());
```

Comment:
Don't we have access to the TX ID somewhere? This way it's nearly impossible to match client-side and Fabric-side logs, which is a bummer for data/log analysis.

Reply:
The SDK has changed now, in the the txID is created mid flow and no longer passed in. It should be possible to pull out the txID at the end and update the completed transaction ... which would yield a similar ability

Reply:
That's fine by me, as long as we have traceability. The Ethereum adapter also sets it at the end :+1: